staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2620,16 +2620,19 @@ type ResourceResizeRestartPolicy string
 
 // These are the valid resource resize restart policy values:
 const (
-	// 'NotRequired' means Kubernetes will try to resize the container
-	// without restarting it, if possible. Kubernetes may however choose to
-	// restart the container if it is unable to actuate resize without a
-	// restart. For e.g. the runtime doesn't support restart-free resizing.
-	NotRequired ResourceResizeRestartPolicy = "NotRequired"
-	// 'RestartContainer' means Kubernetes will resize the container in-place
+	// ResizeRestartPolicyPreferNoRestart means Kubernetes will try to resize the container without
+	// restarting it, if possible. This option does not guarantee that the container won't be
+	// restarted during the resize.
+	// This is the implicit default when no resource resize policy is specified.
+	// +featureGate=InPlacePodVerticalScaling
+	ResizeRestartPolicyPreferNoRestart ResourceResizeRestartPolicy = "PreferNoRestart"
+	// ResizeRestartPolicyNotRequired is equivalent to "PreferNoRestart".
```

Comment:
Can you say more about why it is there and when to use it or not use it?

Reply:
I decided to revert the docs on the `NotRequired` option while PreferNoRestart is in alpha. I'll update this to state that NotRequired is deprecated when PreferNoRestart graduates.